evil-collection.el

```diff
@@ -132,7 +190,9 @@ instance:
       (dolist (req reqs)
         (with-eval-after-load req
           (require (intern (concat "evil-collection-" (symbol-name m))))
-          (funcall (intern (concat "evil-collection-" (symbol-name m) "-setup"))))))))
+          (funcall (intern (concat "evil-collection-" (symbol-name m) "-setup")))
+          (when (fboundp 'evil-collection-user-setup)
+            (funcall #'evil-collection-user-setup m)))))))
```

Comment:
I was thinking this is nice if the user wants to define all their translations in one place, but I'm not sure if *all* users like that kind of configuration.

I could imagine myself putting random swap code in different 'mode' files that I have strewn across my .emacs.d.

I wonder if we want to do something where the user can both define a global function that will run with evil-collection-user-setup as well as a more precise function they can define like evil-collection-specific-mode-user-setup.

The code may just check for one first and run it (or even just run both if both are available).

Reply:
In terms of naming, something like `load-hook` or `setup-hook` would be more conventional, wouldn't it?

I'd expect users to look for a "hook" somewhere, but not necessarily for "user-setup".

We could also use the `run-hooks` function.